eng/source-build-patches/0003-Exclude-some-projects-from-source-build.patch

```diff
@@ -0,0 +1,27 @@
+From 6f903f6870011b6a11074d93a03481faac6a35dc Mon Sep 17 00:00:00 2001
+From: dseefeld <dseefeld@microsoft.com>
+Date: Thu, 4 Feb 2021 15:18:11 +0000
+Subject: [PATCH] Exclude some projects from source-build
+
+---
+ Directory.Build.props | 4 +++-
+ 1 file changed, 3 insertions(+), 1 deletion(-)
+
+diff --git a/Directory.Build.props b/Directory.Build.props
+index 50d30f12fd..85c2f2428b 100644
+--- a/Directory.Build.props
++++ b/Directory.Build.props
+@@ -27,7 +27,9 @@
+         '$(IsTestAssetProject)' == 'true' OR
+         '$(IsBenchmarkProject)' == 'true' OR
+         '$(IsUnitTestProject)' == 'true'">false</IsShipping>
+-
++    <IsOtherTestProject Condition="$(MSBuildProjectName.Contains('IntegrationTesting')) or $(MSBuildProjectName.Contains('TestCommon'))">true</IsOtherTestProject>
++    <ExcludeFromSourceBuild Condition="'$(IsSampleProject)' == 'true' or '$(IsTestAssetProject)' == 'true' or '$(IsBenchmarkProject)' == 'true' or '$(IsUnitTestProject)' == 'true' or '$(IsSpecificationTestProject)' == 'true' or '$(IsOtherTestProject)' == 'true'">true</ExcludeFromSourceBuild>
```

Comment:
We shouldn't need this given https://github.com/dotnet/aspnetcore/blob/607e2164ec4a3fbe020bf3dda578ecb1ffd5f22e/Directory.Build.targets#L11-L19 and existing `'$(IsAspNetCoreApp)' != 'true'` clause in `$(ExcludeFromSourceBuild)` setting a bit above.

Good news is that as long as we use `$(SkipTestBuild)` for source builds, we shouldn't need many (any❔) special cases.